### PR TITLE
post: advanced angular training

### PR DESCRIPTION
**Pull request by cexbrayat**
Blog post announcing our new advanced training.

### CODE REVIEW

**Comment by cexbrayat** [comment]
If you are OK with the content, we'll translate it in french too.

File: _posts/2017-08-10-advanced-angular-training.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: Our Angular Advanced training - from ninja to hero!
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: Ninja to Hero

File: _posts/2017-08-10-advanced-angular-training.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+author: cexbrayat
+tags: ["Angular 2", "Angular", "Angular 4"]
+description: "We have a new training about all the advanced topics: you are going to love it!"
+---
+
+Angular is nearly one year old, and soon in version 5.0!
+
+The first commit is now 3 years old, and we've been using it since the very beginning.
+We have taught the fundamentals of Angular to hundreds of developers in person,
+our [ebook](https://books.ninja-squad.com/angular) have thousands of readers,
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: Start a new sentence: Our ebooks **has**

File: _posts/2017-08-10-advanced-angular-training.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+and our [online training (Pro Pack)](https://angular-exercises.ninja-squad.com/)
+reached a few hundred more developers all over the world!
+
+Along the way we have learn quite a bit about Angular,
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: learnt

File: _posts/2017-08-10-advanced-angular-training.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+its internals, its ecosystem and how to build better applications.
+
+That's why we are really happy to announce **our new 2 days training:
+[Advanced Angular: from ninja to hero](http://ninja-squad.com/formations/formation-angular-advanced)!**
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: Ninja to Hero

File: _posts/2017-08-10-advanced-angular-training.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+This training goes beyond the fundamentals
+and is built for developers who are looking for more advanced topics and best practices.
+
+To give you an overview, the training deep dives in:
+- **dependency injection**: hierarchical injector, why `@Injectable`, singleton vs several instances, how to share data in your application, the different types of providers and their use, `@Optional`, `@Self`, `forwardRef`;
+- **observables**: advanced operators and use-cases, best practices, how to test them;
+- **advanced components and directives**: lifecycle hooks, `ng-content`, `@HostBinging`, `@HostListening`, `ViewChild`, `ContentChild`;
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: HostListener

File: _posts/2017-08-10-advanced-angular-training.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+- **testing**: component tests, how to leverage dependency injection, asynchronous tests;
+- **router**: how to split your application in modules, lazy-loading, the traps to avoid with dependency injection, how to use guards and resolvers;
+- **Angular internals**: how does it work under the hood, change detection, zones, compilation JiT and AoT;
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: JiT and AoT compilation

File: _posts/2017-08-10-advanced-angular-training.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+- **performance**: change detection strategies, element tracking, the `async` pipe and its traps, how to escape Angular's change detection
+- **internationalization**: what's i18n, how to build a multi-language application, best practices
+- **forms**: nested forms, dynamic forms, build your own form component with `ControlValueAccessor`
+- and a lot of tips on how to be ready for **production**, how to set up a **continuous integration**, etc.
+
+This training relies also on our online training,
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: also relies

File: _posts/2017-08-10-advanced-angular-training.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+the [Pro Pack)](https://angular-exercises.ninja-squad.com/)
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: remove the parenthesis after Pack

File: _posts/2017-08-10-advanced-angular-training.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+that you can buy with our [ebook](https://books.ninja-squad.com/angular).
+We'll go over some of the more advanced exercises of the "Pro Pack"
+so you can get your hands dirty with all these new concepts!
+
+Ho, and, as always, the training is **30% off** if you are an independent developer, an open source committer, a conference organizer or if you are currently looking for a job.
```
Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: Ho -> Oh

Comment (jnizet) [inline _posts/2017-08-10-advanced-angular-training.md]: off -> cheaper

**Comment by cexbrayat** [comment]
I found my "solution": I run the blog with ruby 2.3.1 😅 

I've got RVM to manage my ruby versions, and a `.ruby-version` file in the `blog` directory that enforces `2.3.1`.

**Comment by cexbrayat** [comment]
ready to publish on next Thursday

**Comment by cexbrayat** [comment]
Forgot to publish last week. I'll merge today :)

**Pull request merged**